core/src/test/java/hudson/util/AtomicFileWriterTest.java

```diff
@@ -40,12 +31,40 @@
 
 public class AtomicFileWriterTest {
     private static final String PREVIOUS = "previous value \n blah";
+    /**
+     * Provides access to the default permissions given to a new file. (i.e. indirect way to get umask settings).
+     * <p><strong>BEWARE: null on non posix systems</strong></p>
+     */
+    @Nullable
+    private static Set<PosixFilePermission> DEFAULT_GIVEN_PERMISSIONS;
     @Rule
     public TemporaryFolder tmp = new TemporaryFolder();
     File af;
     AtomicFileWriter afw;
     String expectedContent = "hello world";
 
+    @BeforeClass
+    public static void computePermissions() throws IOException {
+        final File tempDir = com.google.common.io.Files.createTempDir();
+        final File newFile = new File(tempDir, "blah");
+        assertThat(newFile.createNewFile(), is(true));
+        if (!isPosixSupported(newFile)) {
+            return;
+        }
+        DEFAULT_GIVEN_PERMISSIONS = Files.getPosixFilePermissions(newFile.toPath());
```

Comment:
:bug: the file and directory are not deleted

Reply:
Not really a bug, granted the tmp directory could grow, but well it's not even production code. Using `TemporaryFolder` instead, constantly forget about this one.

Reply:
@jtnord fixed in last commit.